design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,187 @@
+
+This one size fits all approach is not ideal though, the multiple scenarios and personas need finer grain control around security, access, and isolation.
+This design doc proposes a new approach for the Crossplane Stack Manager (SM) that will support this increased control for the installation of stacks.
+
+## Personas and Scenarios
+
+There are many different scenarios and types of users that may want to install new functionality into their control plane.
+This design proposes support for two high level personas/scenarios as an effective starting point.
+
+* **Platform:** An administrator of the control plane that is installing wide ranging support for lower level infrastructure and platform components,
+such as installing support for an entire cloud provider and all of its infrastructure and platform services.
+  * An example of this would be the `GCP` stack that contains a `Network` primitive and the `CloudSQL` resource.
+This persona would also be expected to define resource classes and other environment specific details.
+* **Application:** An application operator or end user of Crossplane.
+They are focused at a higher level and want to install the functionality of a single application, such as `GitLab` or `WordPress`.
+This persona is not expected to know or control any details about the environments managed by the Crossplane cluster, they simply need to consume their new application of choice.
+
+## Design Summary
+
+The key difference in how the supported personas/scenarios will be handled is in the **scope** of the permissions that they are granted within the control plane.
+
+* **Platform stacks** will be given permissions to watch, access, and create resources at a **cluster level**, i.e. across **all namespaces**.
+* **Application stacks** will be given permissions only within a **single namespace**.
+
+This means that application stacks should not be able to directly affect any resources outside of the specific namespace they are installed into, while platform stacks operate at a higher level of privilege and can affect the entire cluster.
+
+### CRD Scoping
+
+To avoid a common point of confusion, let's review the available scoping of CRDs.
+Custom resource **instances** can be scoped to either a single namespace or to the entire cluster.
+It's important to note that this refers to the instances, not the CRD type definitions.
+From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition):
+
+> CustomResourceDefinitions themselves are non-namespaced and are available to all namespaces
+
+Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
+This has caused confusion in the past, so it's important to clear that up here.
```

Comment:
I appreciate the explicit explanation here. From what I can tell from looking at the diagrams and reading the remainder of this doc, *instances* of every stack-specific CRD kind are namespace-scoped. The only instance where this appears that it may not be the case is `Provider` (see comment below). If it is the case that all instances of CRD's added by a stack should be namespace-scoped, would it be worthwhile to call that out here? The exclusion of it makes me think that we believe there will be uses for both cluster-scoped and namespace-scoped CRD kinds.

Reply:
That was my intent with the Design Summary section that is very near the top, directly after the background section.  I'll add another part to this summary about how this policy is intentionally restrictive initially and that we'll allow real world use cases to influence any expansion to the policy.  I think that summary section will capture everything nicely in the beginning of the doc.